addons/binding/org.openhab.binding.homie/src/main/java/org/openhab/binding/homie/internal/conventionv200/HomieTopic.java

```diff
@@ -0,0 +1,169 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.homie.internal.conventionv200;
+
+import static org.openhab.binding.homie.internal.conventionv200.HomieConventions.*;
+import static org.openhab.binding.homie.internal.conventionv200.TopicParser.*;
+
+import java.util.regex.Matcher;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * This class represents a parsed MQTT homie topic
+ *
+ * @author Michael Kolb - Initial Contribution
+ *
+ */
+public class HomieTopic {
+
+    private final String deviceId;
+    private final String nodeId;
+    private final String subproperty;
+    private final String propertyName;
+    private final String internalPropertyName;
+
+    public HomieTopic(Matcher m) {
+
+        subproperty = m.group(MATCHGROUP_INTERNAL_SUBPROPERTY_NAME);
+        propertyName = m.group(MATCHGROUP_PROPERTY_NAME);
+        internalPropertyName = m.group(MATCHGROUP_INTERNAL_PROPERTY_NAME);
+        deviceId = m.group(MATCHGROUP_DEVICEID_NAME);
+        nodeId = m.group(MATCHGROUP_NODEID_NAME);
+    }
+
+    /**
+     * Check if this topic is a topic published by the device itself
+     *
+     * @return
+     */
+    public boolean isDeviceProperty() {
+        return StringUtils.isBlank(nodeId);
+    }
+
+    /**
+     * Check if this topic is a topic published by a node of a device
+     *
+     * @return
+     */
+    public boolean isNodeProperty() {
+        return StringUtils.isNotBlank(nodeId);
+    }
+
+    /**
+     * Check if this topic is an internal topic (all topics marked with $)
+     *
+     * @return
+     */
+    public boolean isInternalProperty() {
+        return StringUtils.isNotBlank(internalPropertyName);
+    }
+
+    public String getDeviceId() {
+        return deviceId;
+    }
+
+    public String getNodeId() {
+        return nodeId;
+    }
+
+    public String getSubproperty() {
+        return subproperty;
+    }
+
+    public String getPropertyName() {
+        return propertyName;
+    }
+
+    public String getInternalPropertyName() {
+        return internalPropertyName;
+    }
+
+    public boolean isESHNodeUnit() {
+        return isNodeProperty() && !isInternalProperty() && StringUtils.equals(getPropertyName(), ESH_UNIT_TOPIC);
+    }
+
+    /**
```

Comment:
You are not consistent in which methods are documents or is there a reason why you documented `isNodeTypeAnnouncement` and not `isNodePropertyAnnouncement`. I believe the guidelines favor documentation